azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -80,41 +126,55 @@ def _log_policy_event(msg, is_success=True, op=WALAEventOperation.Policy, send_e
     @staticmethod
     def __get_policy_enforcement_enabled():
         """
-        Policy will be enabled if (1) policy file exists at the expected location and (2) the conf flag "Debug.EnableExtensionPolicy" is true.
-        """
-        return conf.get_extension_policy_enabled() and os.path.exists(conf.get_policy_file_path())
+        Return True if policy enforcement is enabled. Policy is enabled if:
+            1. policy file exists at the expected location
+            2. the conf flag "Debug.EnableExtensionPolicy" is set to True.
 
-    @property
-    def policy_enforcement_enabled(self):
-        return self._policy_enforcement_enabled
+        This method runs during policy engine initialization and should not raise errors. If an error occurs, return False.
+        """
+        try:
+            return conf.get_extension_policy_enabled() and os.path.exists(conf.get_policy_file_path())
+        except Exception as ex:
+            logger.warn("Error checking if policy is enabled: {0}".format(ex))
+            return False
```

Comment:
This error handling does not look right. If there is any error, we would not apply policy, even if there is one.

Reply:
IGNORE THIS COMMENT. JUST A REMINDER

[TODO]